c-scape/src/process/exec.rs

```diff
@@ -0,0 +1,200 @@
+use rustix::fd::BorrowedFd;
+use rustix::fs::{AtFlags, Mode, OFlags};
+
+use alloc::vec::Vec;
+use core::ffi::CStr;
+
+use crate::env::environ;
+use crate::{set_errno, Errno};
+
+use libc::{c_char, c_int};
+
+#[no_mangle]
+unsafe extern "C" fn execl(path: *const c_char, arg: *const c_char, mut argv: ...) -> c_int {
+    let mut vec = Vec::new();
+    vec.push(arg);
+
+    loop {
+        let ptr = argv.arg::<*const c_char>();
+        vec.push(ptr);
+        if ptr.is_null() {
+            break;
+        }
+    }
+
+    execv(path, vec.as_ptr())
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn execle(path: *const c_char, arg: *const c_char, mut argv: ...) -> c_int {
```

Comment:
```suggestion
unsafe extern "C" fn execle(path: *const c_char, arg: *const c_char, mut argv: ...) -> c_int {
```
The `pub` here has no immediate effect since it's in a private module, but I want to keep `pub`-vs-not consistent in general.